src/CMakeLists.txt

```diff
@@ -292,7 +292,18 @@ IF (LIBSSH2_FOUND)
 	LIST(APPEND LIBGIT2_PC_LIBS ${LIBSSH2_LDFLAGS})
 	#SET(LIBGIT2_PC_LIBS "${LIBGIT2_PC_LIBS} ${LIBSSH2_LDFLAGS}")
 
-	CHECK_LIBRARY_EXISTS("${LIBSSH2_LIBRARIES}" libssh2_userauth_publickey_frommemory "${LIBSSH2_LIBRARY_DIRS}" HAVE_LIBSSH2_MEMORY_CREDENTIALS)
+	# We might need to provide additional libraries so libssh2 links
+	SET(CMAKE_REQUIRED_LIBRARIES_OLD ${CMAKE_REQUIRED_LIBRARIES})
+	IF(BUILD_SHARED_LIBS)
+		SET(CMAKE_REQUIRED_LIBRARIES ${LIBSSH2_LIBRARIES})
+		SET(LIBSSH2_TMP_LIBRARY_DIRS ${LIBSSH2_LIBRARY_DIRS})
```

Comment:
Do we need this in case of shared libraries? If not, we could probably make this much simpler:

```
IF(BUILD_SHARED_LIBS)
    CHECK_LIBRARY_EXISTS(...) # The old code previous to this commit
ELSE()
    SET(CMAKE_REQUIRED_LIBRARIES ${LIBSSH2_STATIC_LIBRARIES})
    CHECK_LIBRARY_EXISTS(ssh2 libssh2_userauth_publickey_frommemory "${LIBSSH2_STATIC_LIBRARY_DIRS}" HAVE_LIBSSH2_MEMORY_CREDENTIALS)
    UNSET(CMAKE_REQUIRED_LIBRARIES)
ENDIF()

Reply:
True, I was trying to keep a one-and-only-one `CHECK_LIBRARY_EXISTS` call.

Note that I feel like it means something is wrong w.r.t our pkg-config setup in that static-link case (which I'm ignoring because macOS doesn't actually depend on the built pc file). Also, licensing 😟.

Reply:
Yeah, I'm all for avoiding code duplication. But in this case, it's a lot easier to read with the duplicated code, so I'd say in this case duplication is preferable